sdk/core/core-lro/src/poller.ts

```diff
@@ -139,7 +143,7 @@ export abstract class Poller<TState, TResult> {
     }
     if (!this.cancelPromise) {
       this.cancelPromise = this.cancelOnce(options);
-    } else if (options.abortSignal) {
+    } else if (options.abortSignal && options.abortSignal.aborted) {
```

Comment:
I think the reason this is throwing is if you call `cancelOperation` with an abortSignal, we can't hook up your abortSignal to anything, as one is already in progress. In the case where you don't pass anything, we can pass you the pending promise.

So checking `aborted` isn't needed.

Reply:
I'm going to revert the abort change for now and focus just on the promise one. I think I have more questions now for this change.